src/main/java/wedlog/address/model/person/UniqueGuestList.java

```diff
@@ -139,6 +142,26 @@ public int getNumGuestsRsvpUnknown() {
         return numGuestsRsvpUnknown;
     }
 
+    /**
+     * Collates dietary requirements of all guests into a {@code DietaryRequirementStatistics} object.
+     * @return {@code DietaryRequirementStatistics} object containing key-value pairs of dietary requirements.
+     */
+    public DietaryRequirementStatistics getDietaryRequirementStatistics() { // O(n^2) solution. Can optimise?
+        HashMap<String, Integer> dietaryRequirementMap = new HashMap<>();
+        for (Guest guest : internalList) {
+            for (DietaryRequirement dietaryRequirement : guest.getDietaryRequirements()) {
+                String dietaryRequirementString = dietaryRequirement.value;
+                if (dietaryRequirementMap.containsKey(dietaryRequirementString)) {
```

Comment:
maybe we should do toLowerCase on line 153 instead of in the if-else branches. think this line is causing some bugs with the counting right now.